### PR TITLE
Add an argument to CollectionOf()

### PR DESCRIPTION
**Pull request by tchssk**
It makes it possible to specify the collection identifier.

### CODE REVIEW

**Comment by tchssk** [comment]
I don't know why travis was failed :(

File: design/apidsl/media_type.go [hunk 1]
```diff
@@ -376,9 +376,17 @@ func Link(name string, view ...string) {
 // CollectionOf creates a collection media type from its element media type. A collection media
 // type represents the content of responses that return a collection of resources such as "list"
 // actions. This function can be called from any place where a media type can be used.
+//
 // The resulting media type identifier is built from the element media type by appending the media
 // type parameter "type" with value "collection".
-func CollectionOf(v interface{}, apidsl ...func()) *design.MediaTypeDefinition {
+//
+//  CollectionOf(BottleMedia)  // If the identifier of BottleMedia is "vnd.goa.bottle',
+//                             // Content-Type will be "vnd.goa.bottle; type=collection".
```
Comment (raphael) [inline design/apidsl/media_type.go]: Should be `identifier` not `Content-Type` I think. By default the identifier is used to build the content type but that can be overridden with `ContentType`. So maybe something like:
```go
// Examples:
//
//   // Define a collection media type using the default generated identifier
//   // (e.g. "vnd.goa.bottle; type=collection" assuming the identifier of BottleMedia
//   // is "vnd.goa.bottle") and the default views (i.e. inherited from the BottleMedia
//   // views).
//   var col = CollectionOf(BottleMedia)
//
//   // Another collection media type using the same element media type but defining a
//   // a different default view.
//   var col2 = CollectionOf(BottleMedia, "vnd.goa.bottle.alternate; type=collection;", func() {
//        View("default", func() {
//           Attribute("id")
//           Attribute("name")
//        })
//   })
```

File: design/apidsl/media_type.go [hunk 2]
```diff
@@ -376,9 +376,17 @@ func Link(name string, view ...string) {
 // CollectionOf creates a collection media type from its element media type. A collection media
```
Comment (raphael) [inline design/apidsl/media_type.go]: Please remove the first 3 lines:
```go
// CollectionOf can be used in: Wherever a MediaType can be used..
 //
 // e.g. Attribute("foo", CollectionOf(Bar))
```
(I know you didn't add them but they don't add much value and where `MediaType` can be used is already explained below). @deltaskelta fyi - we probably should remove them from the other places that say "wherever...".

Also we should probably change the first line to something like:
```go
// CollectionOf creates a collection media type from its element media type and an optional identifier.

File: design/apidsl/media_type.go [hunk 3]
```diff
@@ -376,9 +376,17 @@ func Link(name string, view ...string) {
...
 // type represents the content of responses that return a collection of resources such as "list"
 // actions. This function can be called from any place where a media type can be used.
+//
 // The resulting media type identifier is built from the element media type by appending the media
```
Comment (raphael) [inline design/apidsl/media_type.go]: Replace this with:
```go
// If an identifier isn't provided then the resulting media type identifier ...
```

File: design/apidsl/media_type.go [hunk 4]
```diff
@@ -376,9 +376,17 @@ func Link(name string, view ...string) {
...
 // type parameter "type" with value "collection".
-func CollectionOf(v interface{}, apidsl ...func()) *design.MediaTypeDefinition {
+//
+//  CollectionOf(BottleMedia)  // If the identifier of BottleMedia is "vnd.goa.bottle',
+//                             // Content-Type will be "vnd.goa.bottle; type=collection".
+//
+// The collection identifier can be specified as second argument.
+//
+//  CollectionOf(BottleMedia, "vnd.goa.bottles")  // Content-Type will be "vnd.goa.bottles".
```
Comment (raphael) [inline design/apidsl/media_type.go]: I guess these last 4 lines are not needed with the example above.

**Comment by tchssk** [comment]
I added a fixing commit!

**Comment by raphael** [comment]
Would you mind backporting this to v1? Thank you!

**Pull request merged**